### PR TITLE
feat: Use cozy-interapp lib 🚜

### PR DESCRIPTION
**Pull request by kosssi**
To remove cozy-client-js we should extract intent in a new lib.
This lib is cozy-interapp, we change instantiation to pass cozy-client

### CODE REVIEW

**Comment by ptbrowne** [comment]
Pourquoi on passe `fetchJSON` et pas le client directement ?

Pourquoi l'application a besoin de savoir que les intents utilisent `fetchJSON` ? IMHO c'est une abtraction qui *leak*.

File: src/components/FileIntentDisplay.jsx [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 import PropTypes from 'prop-types'
+import { Intents } from 'cozy-interapp'
+import { withClient } from 'cozy-client'
+
 import FullscreenIntentModal from 'components/FullscreenIntentModal'
 
 class FileIntentDisplay extends Component {
+  constructor(props) {
+    super(props)
+    this.intents = new Intents({ client: props.client })
```
Comment (kosssi) [inline src/components/FileIntentDisplay.jsx]: New instantiation of Intents

File: test/AppLike.jsx [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 import store from 'test/store'
+import getClient from 'test/client'
 
-const AppLike = ({ children, store }) => (
-  <I18n lang={'en'} dictRequire={() => langEn}>
-    <Provider store={store}>{children}</Provider>
-  </I18n>
+const AppLike = ({ children, store, client }) => (
+  <Provider store={store}>
+    <CozyProvider client={client || getClient()}>
```
Comment (kosssi) [inline test/AppLike.jsx]: Add CozyProvider to have client on context

File: test/client.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import CozyClient from 'cozy-client'
```
Comment (kosssi) [inline test/client.js]: maybe this file needs to be moved in cozy-client

Comment (ptbrowne) [reply test/client.js]: Yep. Though I think that in the tests from Cozy-client, we already have stuff like this where we mock much more. So I think we're going to do this in cozy-client (move code inside cozy-client and export it) and then we're going to be able to import it here. I hope I am understandable.

**Pull request merged**